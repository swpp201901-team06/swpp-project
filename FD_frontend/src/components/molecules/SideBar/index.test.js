import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { SideBar } from './index'

describe('SideBar molecule test', () => {
  const onClickMyArchive = jest.fn()
  const wrap = (props = {}) => shallow(<SideBar {...props} />)

  beforeEach(() => {
    onClickMyArchive.mockClear()
  })

  it('matches snapshot when logged in', () => {
    const props = {
      statefunction: {
        SignInPage: {
          nickname: 'test_nickname',
          isLoggedIn: true,
        },
      },
    }
    const renderedValue = renderer.create(<SideBar {...props} />).toJSON()
    expect(renderedValue).toMatchSnapshot()
  })
})
