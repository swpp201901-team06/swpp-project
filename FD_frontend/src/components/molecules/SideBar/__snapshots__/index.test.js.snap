// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar molecule test matches snapshot when logged in 1`] = `
<div>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="kjvHRn"
      height={40}
      type="button"
    >
      Guest Page
    </button>
  </a>
  <button
    className="kjvHRn"
    height={40}
    onClick={[Function]}
    type="submit"
  >
    Logout
  </button>
  <button
    className="kjvHRn"
    height={40}
    onClick={[Function]}
    type="button"
  >
    My Archive
  </button>
  <a
    onClick={[Function]}
    style={Object {}}
  >
    <button
      className="kjvHRn"
      height={40}
      type="button"
    >
      My Account
    </button>
  </a>
</div>
`;

exports[`SideBar molecule test matches snapshot when logged out 1`] = `
<a
  onClick={[Function]}
  style={Object {}}
>
  <button
    className="kjvHRn"
    height={40}
    type="button"
  >
    Sign In
  </button>
</a>
`;
